### PR TITLE
Make image-user test images multi-arch

### LINKED ISSUE #287: Make test-images multiarch
The "make test-cri" command fails on non-x86 architecture because of the images created from image-test and image-user directory are not multi-arch. If these images are made multi-arch when built and pushed, then it would be easy to test on other non-x86 architectures as well

### PR DESCRIPTION
**Pull request by nitkon**
Fixes #287 
This fixes Issue 287 partially. Here "docker push" is used instead of
"gcloud docker -- push" as its depreciated now. [Reference](https://github.com/kubernetes/kubernetes/commit/c0b73645637c248091358a37c7e99bfcfc983f71)

Signed-off-by: Nitesh Konkar <niteshkonkar@in.ibm.com>

### CODE REVIEW

File: images/image-user/Makefile [hunk 1]
```diff
@@ -12,22 +12,33 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY: all test-image-user-uid test-image-user-username test-image-user-uid-group test-image-user-username-group
+include ../../hack/make-rules/Makefile.manifest
+
+.PHONY: all test-image-user-uid test-image-user-username test-image-user-uid-group test-image-user-username-group push-manifest
+
+REGISTRY = gcr.io/cri-tools
+ALL_ARCH = amd64 ppc64le
```
Comment (feiskyer) [inline images/image-user/Makefile]: Add all arches that kubernetes has already supported? e.g. `amd64 arm arm64 ppc64le s390x`

File: images/image-user/Dockerfile [hunk 2]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM busybox
+ARG ARCH
+FROM $ARCH/busybox
```
Comment (feiskyer) [inline images/image-user/Dockerfile]: seems there are no `arm/busybox` and `arm64/busybox` images?

File: images/image-user/Makefile (hunk 1 above)
Comment (nitkon) [reply images/image-user/Makefile]: @feiskyer : Should we use arm32v7 and arm64v8 ? [Reference](https://github.com/docker-library/official-images#architectures-other-than-amd64)

File: images/image-user/Dockerfile (hunk 2 above)
Comment (nitkon) [reply images/image-user/Dockerfile]: @feiskyer : Should we use arm32v7 and arm64v8 ? [Reference](https://github.com/docker-library/official-images#architectures-other-than-amd64)

Comment (mkumatag) [reply images/image-user/Dockerfile]: yes we have to use `arm32v7` and `arm64v8`

**Comment by nitkon** [comment]
@feiskyer : @mkumatag :  Addressed your comments.

File: images/image-user/Makefile [hunk 3]
```diff
@@ -12,22 +12,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+include ../../hack/make-rules/Makefile.manifest
+include ../../hack/make-rules/Baseimage
```
[the remark this answers is not in the document]
Comment (nitkon) [reply images/image-user/Makefile]: @feiskyer :  Thnx! Fixed.

**Comment by feiskyer** [comment]
thanks

/lgtm

**Comment by feiskyer** [comment]
@Random-Liu Could you build and push those new images?

**Comment by mkumatag** [comment]
It depends on the `.docker/config` file entry with proper credentials to login `gcr.io`. So you need to check whether docker command is able to push the image to gcr or not.

**Comment by nitkon** [comment]
@Random-Liu : I have addressed your comment in the following [PR](https://github.com/kubernetes-incubator/cri-tools/pull/311)

**Comment by Random-Liu** [comment]
> It depends on the .docker/config file entry with proper credentials to login gcr.io. So you need to check whether docker command is able to push the image to gcr or not.

I did `gcloud credential` thing, let me try again tonight.
